src/masternode/payments.h

```diff
@@ -37,6 +39,7 @@ class MNSubsidyAgent
     const Consensus::Params& m_consensus_params;
     const CMasternodeSync& m_mn_sync;
     const CSporkManager& m_sporkman;
+    const std::unique_ptr<CDeterministicMNManager>& m_dmnman;
```

Comment:
before deglobalization of CDeterministicMNManager here I think we should firstly break this circular dependency.

Reply:
`CDeterministicMNManager` is quite extensively used in the codebase, even with two commits (split for ease of review), there are at least 23 uses (as of 26a11d75643f2401cab4c3f82bd237157a9f30d5) that need to be passed as arguments, which balloons to more changes especially if it's an independent function or part of a class that is expected to have no arguments in its constructor.

![Twenty three uses of deterministicMNManager](https://github.com/dashpay/dash/assets/63189531/8f7e4791-10cc-45c1-abca-8503319ba703)

The most concerning one being in `CTxMemPool`, which was migrated to ref-by-arg but found to break tests and was excluded to allow the current set of changes to be reviewed.

But 23 is an improvement over 147 (as of 2f6528292fe4d76b9a9916b26af13ba0bc223457) and so I propose these set of changes remain. Tackling that circular dependency will need to be addressed at some point but I wouldn't stall improvements for it.

![One hundred forty seven uses of deterministicMNManager](https://github.com/dashpay/dash/assets/63189531/b59e3254-18d8-4329-89d1-1ad982e5de29)